buffer/ring_growing.go

```diff
@@ -34,19 +34,14 @@ func NewRingGrowing(initialSize int) *RingGrowing {
 }
 
 // ReadOne reads (consumes) first item from the buffer if it is available, otherwise returns false.
-func (r *RingGrowing) ReadOne() (data interface{}, ok bool) {
+func (r *RingGrowing) ReadOne() (interface{}, bool) {
 	if r.readable == 0 {
 		return nil, false
 	}
 	r.readable--
 	element := r.data[r.beg]
 	r.data[r.beg] = nil // Remove reference to the object to help GC
-	if r.beg == r.n-1 {
-		// Was the last element
-		r.beg = 0
-	} else {
-		r.beg++
-	}
+	r.beg = (r.beg + 1) % r.n
```

Comment:
This change takes a simple adder operation and makes it a divide.  I wouldn't pay the additional cost to make the code smaller.